airflow/contrib/utils/input_validator.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from airflow import AirflowException
+
+
+class InputValidationMixin(object):
+
+    REQUIRED_ATTRIBUTES = []
+    REQUIRED_NOT_EMPTY_ATTRIBUTES = []
+
+    def _validate_inputs(self):
+        for attr_name in self.REQUIRED_ATTRIBUTES:
+            if not getattr(self, attr_name):
+                raise AirflowException("The required parameter '{}' is empty".format(attr_name))
```

Comment:
I think we should say "is None" here rather than "is empty". If you also iterate over both arrays in the next step then we could check two conditions for REQUIRED_ATTRIBUTE. In this line we check if it is in the dictionary and in the next one we check if is not == "".